mw_jwt.go

```diff
@@ -206,22 +225,41 @@ func (k *JWTMiddleware) getBasePolicyID(token *jwt.Token) (string, bool) {
 	return "", false
 }
 
+func (k *JWTMiddleware) getUserIdFromClaim(token *jwt.Token) (string, error) {
+	var userId string
+	var found bool
+
+	if k.Spec.JWTIdentityBaseField != "" {
+		userId, found = token.Claims.(jwt.MapClaims)[k.Spec.JWTIdentityBaseField].(string)
+
+		if !found {
+			log.Warning("Base Field not found, falling back to SUB")
+		}
+	}
+	if !found {
+		userId, found = token.Claims.(jwt.MapClaims)[SUB].(string)
```

Comment:
just a general comment about readability - i am kind of worried about expressions containing several asserts and access to map key, hard to read.

also, current code should work, but `found` is not like real sign that sub was found in claims - it is more it was found in claims and it was a string

Reply:
Have fixed it.